### DISCUSSION TITLE
How to calculate the checksum for ICMPv6 protocol during fuzz testing?

### DISCUSSION THREAD

**Discussion by newbing9527** [jtpereyda/boofuzz#716] [Q&A]
boofuzz RawL2SocketConnection fuzz testing for ICMPv6 protocol, including the checksum part with the IPv6 pseudo-header (reference: RFC2460), as well as the entire ICMPv6 field. How can we dynamically calculate the checksum during fuzz testing?

**Comment by SR4ven** [comment]
There is a [checksum](https://github.com/jtpereyda/boofuzz/blob/master/boofuzz/blocks/checksum.py) block which you can pass a block name of which to calculate the checksum of, as well as a custom algorithm.
Hope that helps.

**Reply by newbing9527** [reply]
From the source code, it looks that this method does not support the calculation of IPv6-related checksums.
